libs/dyn/yamlloader/yaml_spec_test.go

```diff
@@ -801,6 +822,7 @@ func TestYAMLSpecExample_2_27(t *testing.T) {
 func TestYAMLSpecExample_2_28(t *testing.T) {
 	file := "testdata/spec_example_2.28.yml"
 	self := loadExample(t, file)
+	self = self.DropKeyLocations()
```

Comment:
Could you include one or two tests that confirm that map keys actually have a location?

Requires a bit more work on constructing the expected value, but worth it to include it. Can be a follow up.